source/common/quic/platform/quiche_logging_impl.h

```diff
@@ -6,4 +6,267 @@
+  QUICHE_CHECK_INNER_IMPL((a)op(b),                                                                \
+                          "CHECK failed: " #a " (=" << (a) << ") " #op " " #b " (=" << (b) << ")")
+
+#define QUICHE_CHECK_GT_IMPL(a, b) QUICHE_CHECK_INNER_IMPL_OP(>, a, b)
+#define QUICHE_CHECK_GE_IMPL(a, b) QUICHE_CHECK_INNER_IMPL_OP(>=, a, b)
+#define QUICHE_CHECK_LT_IMPL(a, b) QUICHE_CHECK_INNER_IMPL_OP(<, a, b)
+#define QUICHE_CHECK_LE_IMPL(a, b) QUICHE_CHECK_INNER_IMPL_OP(<=, a, b)
+#define QUICHE_CHECK_NE_IMPL(a, b) QUICHE_CHECK_INNER_IMPL_OP(!=, a, b)
+#define QUICHE_CHECK_EQ_IMPL(a, b) QUICHE_CHECK_INNER_IMPL_OP(==, a, b)
+
+#ifdef NDEBUG
+// Release build
+#define QUICHE_DCHECK_IMPL(condition) QUICHE_COMPILED_OUT_LOG(condition)
+#define QUICHE_COMPILED_OUT_LOG(condition)                                                         \
+  QUICHE_LOG_IMPL_INTERNAL(false && (condition), quic::NullLogStream().stream())
+#define QUICHE_DVLOG_IMPL(verbosity) QUICHE_COMPILED_OUT_LOG(false)
+#define QUICHE_DVLOG_IF_IMPL(verbosity, condition) QUICHE_COMPILED_OUT_LOG(condition)
+#define QUICHE_DLOG_IMPL(severity) QUICHE_COMPILED_OUT_LOG(false)
+#define QUICHE_DLOG_IF_IMPL(severity, condition) QUICHE_COMPILED_OUT_LOG(condition)
+#define QUICHE_DLOG_INFO_IS_ON_IMPL() 0
+#define QUICHE_DLOG_EVERY_N_IMPL(severity, n) QUICHE_COMPILED_OUT_LOG(false)
+#define QUICHE_NOTREACHED_IMPL()
+#define QUICHE_DCHECK_GT_IMPL(a, b) QUICHE_COMPILED_OUT_LOG((a) > (b))
+#define QUICHE_DCHECK_GE_IMPL(a, b) QUICHE_COMPILED_OUT_LOG((a) >= (b))
+#define QUICHE_DCHECK_LT_IMPL(a, b) QUICHE_COMPILED_OUT_LOG((a) < (b))
+#define QUICHE_DCHECK_LE_IMPL(a, b) QUICHE_COMPILED_OUT_LOG((a) <= (b))
+#define QUICHE_DCHECK_NE_IMPL(a, b) QUICHE_COMPILED_OUT_LOG((a) != (b))
+#define QUICHE_DCHECK_EQ_IMPL(a, b) QUICHE_COMPILED_OUT_LOG((a) == (b))
+#else
+// Debug build
+#define QUICHE_DCHECK_IMPL(condition) QUICHE_CHECK_IMPL(condition)
+#define QUICHE_DVLOG_IMPL(verbosity) QUICHE_VLOG_IMPL(verbosity)
+#define QUICHE_DVLOG_IF_IMPL(verbosity, condition) QUICHE_VLOG_IF_IMPL(verbosity, condition)
+#define QUICHE_DLOG_IMPL(severity) QUICHE_LOG_IMPL(severity)
+#define QUICHE_DLOG_IF_IMPL(severity, condition) QUICHE_LOG_IF_IMPL(severity, condition)
+#define QUICHE_DLOG_INFO_IS_ON_IMPL() QUICHE_LOG_INFO_IS_ON_IMPL()
+#define QUICHE_DLOG_EVERY_N_IMPL(severity, n) QUICHE_LOG_EVERY_N_IMPL(severity, n)
+#define QUICHE_NOTREACHED_IMPL() PANIC("reached unexpected code")
+#define QUICHE_DCHECK_GT_IMPL(a, b) QUICHE_CHECK_GT_IMPL(a, b)
+#define QUICHE_DCHECK_GE_IMPL(a, b) QUICHE_CHECK_GE_IMPL(a, b)
+#define QUICHE_DCHECK_LT_IMPL(a, b) QUICHE_CHECK_LT_IMPL(a, b)
+#define QUICHE_DCHECK_LE_IMPL(a, b) QUICHE_CHECK_LE_IMPL(a, b)
+#define QUICHE_DCHECK_NE_IMPL(a, b) QUICHE_CHECK_NE_IMPL(a, b)
+#define QUICHE_DCHECK_EQ_IMPL(a, b) QUICHE_CHECK_EQ_IMPL(a, b)
+#endif
+
+#define QUICHE_PREDICT_FALSE_IMPL(x) ABSL_PREDICT_FALSE(x)
+#define QUICHE_PREDICT_TRUE_IMPL(x) ABSL_PREDICT_TRUE(x)
+
+namespace quic {
```

Comment:
s/quic/quiche
Is this file moved from source/common/quic/platform/quic_logging_impl.h? Using git mv would have show the diff.

Reply:
Nice catch about "quic". There were quite a few other such terms and I renamed those too, now.

I did use git mv, but I think it's complicated by the fact that tip of tree already had a quiche_logging_impl.h, but it was just #include quic_logging_impl.h, as I mentioned in the description. I think this is preventing the move detection from working correctly :(